flink-runtime/src/main/java/org/apache/flink/runtime/clusterframework/BootstrapTools.java

```diff
@@ -640,4 +641,30 @@ public static String getDynamicProperties(Configuration baseConfig, Configuratio
 			.toArray(String[]::new);
 		return String.join(" ", newAddedConfigs);
 	}
+
+	/**
+	 * Calculate heap size after cut-off.
+	 */
+	public static int calculateHeapSize(int memory, org.apache.flink.configuration.Configuration conf) {
```

Comment:
I am not quite sure whether this function is out of scope of the class. From the definition of this class, it should not cover this function. Can you find other proper places for this utility or we can define a new util class for it?

Reply:
nit: can remove the prefix `org.apache.flink.configuration`

Reply:
Currently, the only usage of calculating cutoff heap size is to set `-Xmx -Xms` for jobmanager start command. So i intend to update the description of this method to reflect this. When we refactor the memory configuration for jobmanager based on FLIP-49, we could  reconsider whether we need a separate class for jobmanager resource utils.

Reply:
Make sense for me.